k9mail/src/main/java/com/fsck/k9/controller/MessagingController.java

```diff
@@ -3517,7 +3527,7 @@ public void deleteDraft(final Account account, long id) {
             if (uid != null) {
                 LocalMessage message = localFolder.getMessage(uid);
                 if (message != null) {
-                    deleteMessages(Collections.singletonList(message), null);
+                    deleteMessages(Collections.singletonList(message.makeMessageReference()), null);
```

Comment:
I'm not entirely sure what you wanted here, but I noticed the getMessage() call is unnecessary since we can just directly get a MessageReference from the info we have from before. Pushed some changes that might cover what you said as well